### PR TITLE
Enable hdfs cros

### LINKED ISSUE #2551: Will hdfs support CORS?
## TL;DR
Some features in webportal need ajax requests to hdfs, which will cause a CORS problem. So will the hdfs be configured to support CORS strategies?

## Details
For example, if we want to support user upload local data to hdfs from web portal or auto fill when user inputing a hdfs path, there will be a realtime request to list the paths in hdfs.

Since webportal and hdfs are deployed in different origins, the request from webportal to hdfs will be a CORS.

From the [official doc](https://hadoop.apache.org/docs/r2.7.2/hadoop-project-dist/hadoop-common/HttpAuthentication.html#CORS), we can set hdfs config to enable the http request from webportal.
 
If we don't set CORS from hdfs, the browser will block all cors request for security, hence we have to disable all the features that needs it.

Note:
If we request through pylon, there will be no cors. However we have to support general condition without pylon.

### PR DESCRIPTION
**Pull request by mzmssg**
Fix #2551 
According to https://hadoop.apache.org/docs/r2.9.0/hadoop-project-dist/hadoop-common/HttpAuthentication.html#CORS

### CODE REVIEW

**Comment by debuggy** [comment]
- Maybe we should test this feature on a test bench when merging into master. 
- And also we should consider all the cors scenarios and make the security risk to be lowest.

File: src/hadoop-data-node/deploy/hadoop-data-node-configuration/core-site.xml [hunk 1]
```diff
@@ -17,27 +17,66 @@
...
+  </property>
+
+  <property>
+    <name>hadoop.http.cross-origin.enabled</name>
+    <value>true</value>
+  <description>Enables cross origin support for all web-services</description>
+  </property>
+
+  <property>
+    <name>hadoop.http.cross-origin.allowed-origins</name>
```
[the remark this answers is not in the document]
Comment (mzmssg) [reply src/hadoop-data-node/deploy/hadoop-data-node-configuration/core-site.xml]: No much need to complicate it , our hdfs is in insecure mode itself.

File: src/hadoop-data-node/deploy/hadoop-data-node-configuration/core-site.xml [hunk 2]
```diff
@@ -17,27 +17,66 @@
...
+    <value>*</value>
+    <description>Comma separated list of origins that are allowed, wildcards (*) and patterns allowed</description>
+  </property>
+
+  <property>
+    <name>hadoop.http.cross-origin.allowed-methods</name>
+    <value>GET,POST,HEAD,PUT,OPTIONS,DELETE</value>
```
[the remark this answers is not in the document]
Comment (mzmssg) [reply src/hadoop-data-node/deploy/hadoop-data-node-configuration/core-site.xml]: Sure, I will remove it if webportal don't neet it.

File: src/hadoop-data-node/deploy/hadoop-data-node-configuration/core-site.xml [hunk 3]
```diff
@@ -17,27 +17,66 @@
...
+
+  <property>
+    <name>hadoop.http.cross-origin.allowed-methods</name>
+    <value>GET,POST,HEAD,PUT,OPTIONS</value>
+    <description>Comma separated list of methods that are allowed</description>
+  </property>
+
+  <property>
+    <name>hadoop.http.cross-origin.allowed-headers</name>
+    <value>X-Requested-With,Content-Type,Accept,Origin,WWW-Authenticate,Accept-Encoding,Transfer-Encoding</value>
```
Comment (Gerhut) [inline src/hadoop-data-node/deploy/hadoop-data-node-configuration/core-site.xml]: Per <https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#Simple_requests>, I think currently

```suggestion
    <value>X-Requested-With,Content-Type,Origin</value>
```

is OK, more allowed headers introduce more risks

**Comment by mzmssg** [comment]
@debuggy 
I do a simple test on webhdfs api, cors header has been added, please help add more test cases if necessary:
```
> GET /webhdfs/v1/yarn?op=GETFILESTATUS HTTP/1.1
> Host: xx.xx.xx.xx:5070
> User-Agent: curl/7.60.0
> Accept: */*
> Origin: http://example.com
>
< HTTP/1.1 200 OK
< Cache-Control: no-cache
< Expires: Mon, 15 Apr 2019 06:43:09 GMT
< Date: Mon, 15 Apr 2019 06:43:09 GMT
< Pragma: no-cache
< Expires: Mon, 15 Apr 2019 06:43:09 GMT
< Date: Mon, 15 Apr 2019 06:43:09 GMT
< Pragma: no-cache
< Content-Type: application/json
< X-FRAME-OPTIONS: SAMEORIGIN
< Access-Control-Allow-Origin: http://example.com
< Access-Control-Allow-Credentials: true
< Access-Control-Allow-Methods: GET,POST,HEAD,PUT,OPTIONS
< Access-Control-Allow-Headers: X-Requested-With,Content-Type,Origin
< Access-Control-Max-Age: 1800
< Transfer-Encoding: chunked
```

**Pull request merged**